test/e2e/common/util_cleanup.go

```diff
@@ -6,95 +6,136 @@ import (
-	logf.Log.Info("DeleteAllVolumeResources")
-	foundResources := false
-	success := true
+// returns ok -> operations succeeded, resources undeleted, delete resources failed
+func DeleteAllPvcs() (int, error) {
+	logf.Log.Info("DeleteAllPvcs")
 
 	// Delete all PVCs found
-	// Phase 1 to delete dangling resources
 	pvcs, err := gTestEnv.KubeInt.CoreV1().PersistentVolumeClaims("default").List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		logf.Log.Info("DeleteAllVolumeResources: list PVCs failed.", "error", err)
-		success = false
-	}
-	if err == nil && pvcs != nil && len(pvcs.Items) != 0 {
-		foundResources = true
-		logf.Log.Info("DeleteAllVolumeResources: deleting PersistentVolumeClaims")
+		logf.Log.Info("DeleteAllPvcs: list PersistentVolumeClaims failed.", "error", err)
+	} else if len(pvcs.Items) != 0 {
 		for _, pvc := range pvcs.Items {
-			if err := DeletePVC(pvc.Name); err != nil {
-				success = false
+			logf.Log.Info("DeleteAllPvcs: deleting", "PersistentVolumeClaim", pvc.Name)
+			delErr := gTestEnv.KubeInt.CoreV1().PersistentVolumeClaims("default").Delete(context.TODO(), pvc.Name, metav1.DeleteOptions{GracePeriodSeconds: &ZeroInt64})
+			if delErr != nil {
+				logf.Log.Info("DeleteAllPvcs: failed to delete", "PersistentVolumeClaim", pvc.Name, "error", delErr)
 			}
 		}
 	}
 
+	// Wait 2 minutes for PVCS to be deleted
+	numPvcs := 0
+	for attempts := 0; attempts < 120; attempts++ {
+		pvcs, err := gTestEnv.KubeInt.CoreV1().PersistentVolumeClaims("default").List(context.TODO(), metav1.ListOptions{})
+		if err == nil {
+			numPvcs = len(pvcs.Items)
+			if numPvcs == 0 {
+				break
+			}
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	logf.Log.Info("DeleteAllPvcs:", "number of PersistentVolumeClaims", numPvcs, "error", err)
+	return numPvcs, err
+}
+
+func DeleteAllPvs() (int, error){
 	// Delete all PVs found
+	// First remove all finalizers
 	pvs, err := gTestEnv.KubeInt.CoreV1().PersistentVolumes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		logf.Log.Info("DeleteAllVolumeResources: list PVs failed.", "error", err)
-	}
-	if err == nil && pvs != nil && len(pvs.Items) != 0 {
-		logf.Log.Info("DeleteAllVolumeResources: deleting PersistentVolumes")
+		logf.Log.Info("DeleteAllPvs: list PersistentVolumes failed.", "error", err)
+	} else if len(pvs.Items) != 0 {
+		empty := make([]string, 0)
 		for _, pv := range pvs.Items {
-			if err := gTestEnv.KubeInt.CoreV1().PersistentVolumes().Delete(context.TODO(), pv.Name, metav1.DeleteOptions{}); err != nil {
-				success = false
+			finalizers := pv.GetFinalizers()
+			if len(finalizers) != 0 {
+				logf.Log.Info("DeleteAllPvs: deleting finalizer for",
+					"PersistentVolume", pv.Name, "finalizers", finalizers)
+				pv.SetFinalizers(empty)
+				_, _ = gTestEnv.KubeInt.CoreV1().PersistentVolumes().Update(context.TODO(), &pv, metav1.UpdateOptions{})
 			}
 		}
 	}
 
-	// Wait 2 minutes for resources to be deleted
+	// then Wait 1 minute for resources to be cleared
```

Comment:
We are waiting for a maximum of 2 minutes here.

Reply:
Done